### PR TITLE
PWA-1850: Add to Cart on CMS - Out of stock is visible

### PR DESCRIPTION
**Pull request by michaelyu0123**
- Updated pagebuilder products to use url key instead of sku for getting products to fetch out of stock products

## Description
Update template to properly show out of stock item in a CMS context.

## Related Issue

Closes: [[PWA-1850] Add to Cart on CMS - Out of stock is visible](https://jira.corp.magento.com/browse/PWA-1850)

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### CODE REVIEW

**Review by JustinConabreeAbsolunet — CHANGES_REQUESTED** [review]
Very minor change request, and hopefully there's a way to use a less specific selector so it's less easy to break

File: packages/pagebuilder/lib/ContentTypes/Products/products.js [hunk 1]
```diff
@@ -81,8 +83,24 @@ const Products = props => {
...
+        nextFetchPolicy: 'cache-first'
+    });
+
+    const productUrlSuffix = useMemo(() => {
+        if (storeConfigData) {
+            return storeConfigData.storeConfig.product_url_suffix;
+        }
+    }, [storeConfigData]);
+
+    const urlKeys = [...pathNames].map(pathName => {
```
[the remark this answers is not in the document]
Comment (michaelyu0123) [reply packages/pagebuilder/lib/ContentTypes/Products/products.js]: updated

File: packages/pagebuilder/lib/ContentTypes/Products/configAggregator.js [hunk 2]
```diff
@@ -1,7 +1,9 @@
 import { getAdvanced } from '../../utils';
 
 export default (node, props) => {
-    const forms = node.querySelectorAll('[data-product-sku]');
+    const forms = node.querySelectorAll(
+        '.product-item-details > .product-item-name > a.product-item-link'
```
[the remark this answers is not in the document]
Comment (JustinConabreeAbsolunet) [reply packages/pagebuilder/lib/ContentTypes/Products/configAggregator.js:5]: Letting this one go since we need to ensure we only get one match per item.

**Review by JustinConabreeAbsolunet — APPROVED** [review]

**Comment by mikhaelbois** [comment]
@michaelyu0123 
Product carrousel no longer showing on home page.
There are no errors in the console.
Those products are in stock and should not have been impacted by this PR.
I have also tested it on a local environment with the **Display Out of Stock Products** enabled and the products are not shown.
We only get an empty slide div.

**Comment by michaelyu0123** [comment]
> @michaelyu0123 Product carrousel no longer showing on home page. There are no errors in the console. Those products are in stock and should not have been impacted by this PR. I have also tested it on a local environment with the **Display Out of Stock Products** enabled and the products are not shown. We only get an empty slide div.

Resolved. It should show up now.

**Review by JustinConabreeAbsolunet — APPROVED** [review]
LGTM!

**Comment by mikhaelbois** [comment]
@michaelyu0123 
Great, the slider is now showing again.
I'll do some testing on my local for the new features.

Also, coverage is slightly impacted. If it's not too much work, would it be possible to check that out?

**Pull request merged**